app/src/main/java/itaycsguy/rtchordslearningapk/SigninActivity.kt

```diff
@@ -14,140 +14,166 @@ import com.google.android.gms.auth.api.signin.GoogleSignInResult
-        buttonSignIn.setOnClickListener {
-            val username = (findViewById(R.id.text_username) as EditText).text.toString()
-            val password = (findViewById(R.id.text_password) as EditText).text.toString()
-            this.local_validate(username,password)
+        this.username = findViewById(R.id.text_username) as EditText
+        this.password = findViewById(R.id.text_password) as EditText
+        this.localSignInButton = findViewById(R.id.sign_in_button) as Button
+        this.localSignUpButton = findViewById(R.id.sign_up_button) as Button
+        this.localForgotButton = findViewById(R.id.forgot_button) as Button
+        this.signInButton = findViewById(R.id.google_sign_in) as SignInButton
+        this.signInButton.setOnClickListener(this)
+        this.googleApiClient = GoogleApiClient.Builder(this).enableAutoManage(this,this).addApi(Auth.GOOGLE_SIGN_IN_API,this.signInOptions).build()
+        this.localSignInButton.setOnClickListener {
+            this.text_username = this.username.text.toString()
+            this.localValidate(this.username.text.toString(),this.password.text.toString())
         }
-        val buttonSignUp = findViewById(R.id.sign_up_button) as Button
-        buttonSignUp.setOnClickListener {
+        this.localSignUpButton.setOnClickListener {
             val intent = Intent(this, SignupActivity::class.java)
             startActivity(intent)
         }
-        val forgotButton = findViewById(R.id.forgot_button) as Button
-        forgotButton.setOnClickListener {
+        this.localForgotButton.setOnClickListener {
             val intent = Intent(this, SigninRecoveryActivity::class.java)
             startActivity(intent)
         }
     }
 
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+        this.init()
+        if(intent.hasExtra("sign_out")){
+            val isSignedOut = intent.getBooleanExtra("sign_out",false)
+            if (isSignedOut) {
+//                temporary msg:
+                this.signOut()
+//                this.localSignOut() // need still to implement
+            }
+        } else if(intent.hasExtra("username") && intent.hasExtra("password")){
+            val username = intent.getCharSequenceExtra("username").toString()
+            val password = intent.getCharSequenceExtra("password").toString()
+            this.fillDetails(username,password)
+
+        }
+    }
+
+    private fun fillDetails(username: String,password: String){
+        this.username.setText(username)
+        this.password.setText(password)
+    }
+
     override fun onClick(p0: View) {
-//        need to check whether the 'id' of p0 is the same as for google button on the UI
         when(p0.id){
             R.id.google_sign_in -> this.signIn()
-            R.id.google_sign_out -> this.signOut()
+//            R.id.google_sign_out -> this.signOut()
```

Comment:
remove all unused code( comments "//")

Reply:
done!
It is presenting the  "on the flow" classes design.